src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/domain/UnallocatedCaseRisks.kt

```diff
@@ -59,41 +49,59 @@ data class UnallocatedCaseRisks @JsonCreator constructor(
 
 data class UnallocatedCaseRegistration @JsonCreator constructor(
   @Schema(description = "Type", example = "Suicide/self-harm")
-  val type: String,
+  val type: String?,
   @Schema(description = "Registered date", example = "2020-03-21")
   @JsonFormat(pattern = "yyyy-MM-dd", shape = JsonFormat.Shape.STRING)
-  val registered: LocalDate,
+  val registered: LocalDate?,
   @Schema(description = "Notes", example = "Previous suicide /self-harm attempt. Needs further investigating.")
   val notes: String?,
   @Schema(description = "End Date", example = "2020-01-16")
   @JsonFormat(pattern = "yyyy-MM-dd", shape = JsonFormat.Shape.STRING)
   val endDate: LocalDate?,
 ) {
   companion object {
-    fun from(offenderRegistration: OffenderRegistration): UnallocatedCaseRegistration {
+    fun from(registrations: Registrations): UnallocatedCaseRegistration {
       return UnallocatedCaseRegistration(
-        offenderRegistration.type.description,
-        offenderRegistration.startDate,
-        offenderRegistration.notes,
-        offenderRegistration.endDate
+        registrations.description,
+        registrations.startDate,
+        registrations.notes,
+        registrations.endDate
       )
     }
   }
 }
 
 data class UnallocatedCaseRsr @JsonCreator constructor(
   @Schema(description = "Level", example = "HIGH")
-  val level: String,
+  val level: String?,
   @Schema(description = "last updated on Date", example = "2020-01-16")
   @JsonFormat(pattern = "yyyy-MM-dd", shape = JsonFormat.Shape.STRING)
-  val lastUpdatedOn: LocalDate,
-  val percentage: BigDecimal
-)
+  val lastUpdatedOn: LocalDate?,
+  val percentage: BigDecimal?
+) {
+  companion object {
+    fun from(rp: RiskPredictor?): UnallocatedCaseRsr? {
+      return UnallocatedCaseRsr(
```

Comment:
I think `rp?.let{}` is a bit better here otherwise you're making everything in the target class optional when it may not be